### DISCUSSION TITLE
Poll multiple IP addresses with one instance

### DISCUSSION THREAD

**Discussion by dakhnod** [gavinying/modpoll#63] [Ideas]
Is it possible to have one instance of modpoll running which polls from multiple different IP addresses, just as it can poll from different modbus addresses?

**Comment by gavinying** [comment]
Hi Daniel, you have to run multiple modpoll processes to poll from different IP addresses in current version, but thanks for your suggestion. I know this is a quite useful feature, we also had a similar discussion last year, see https://github.com/gavinying/modpoll/issues/11

Previously, the `modpoll` tool was primarily positioned as an ad-hoc troubleshooting tool, with the design focused on the following key aspects:

- **Reusability**: The CSV configuration file contains only vendor-defined device register information, ensuring it can be shared freely without privacy concerns.
- **Flexibility**: In addition to the configuration file, modpoll provides numerous command-line arguments to support various customer-defined network configurations (including IP addresses and server URLs).
- **Deployability**: For small-scale deployments, Docker and Docker Compose help manage network layer configurations.

So far, performance and scalability were not the focus in previous releases.

However, I’ve noticed an increasing demand for using `modpoll` in large-scale deployments recently. Requests to improve network configuration handling and overall usability are growing. I am considering revisiting the design and incorporating additional features to meet these needs. Stay tuned.

BTW, all suggestions are welcome.
